### DISCUSSION TITLE
What is the intended way to do anything? (Noob)

### DISCUSSION THREAD

**Discussion by Ciel-MC** [mrcjkb/haskell-tools.nvim#305] [Q&A]
I am new to haskell in general so please forgive any ignorance, but what is the intended way to do things

I wrote a function and would like to test it (with the repl), how do I do that?

And if I want to formalize it, what about unit tests?

And if my program once done and I want to start at main, how do I run it from inside NVim? I don't see a `:HSRun` command, do I need a project to do this? `Cobal init` is too long, I'm just trying to do CoA, but without a cobal/stack project I can't seem to be able to get dependencies(or maybe I do? Idk how to run it, but the diagnostic for missing lib is present even though ghci lets me import it).

Lastly, what is this code lens thing that appears as a comment, what is it even running? Unit tests?

Sorry this turned into a ramble but everything is so confusing, and I can't seem to find the proper documentation anywhere :sob:, everything just assumes you already know what you want, but I don't.

**Comment by mrcjkb** [comment]
Hey 👋 

Most of this plugin's features that you would be interested in as a Haskell beginner are built around haskell-language-server.

For that, I would say the best place to start are the [haskell-language-server docs](https://haskell-language-server.readthedocs.io/en/latest/).

For learning Haskell in general, I think it may be better to go without haskell-language-server to begin with, and to follow your learning resource's tutorials using either the Stack or Cabal CLI, or even plain GHC/GHCi.
